src/Gui/View3DInventorViewer.cpp

```diff
@@ -567,11 +567,15 @@ void View3DInventorViewer::init()
     //filter a few qt events
     viewerEventFilter = new ViewerEventFilter;
     installEventFilter(viewerEventFilter);
+#if defined(Q_OS_MACOS) || defined(Q_OS_WIN32)
     ParameterGrp::handle hViewGrp = App::GetApplication().GetParameterGroupByPath(
         "User parameter:BaseApp/Preferences/View");
     if (hViewGrp->GetBool("LegacySpaceMouseDevices", false)) {
         getEventFilter()->registerInputDevice(new SpaceNavigatorDevice);
     }
+#else
+    getEventFilter()->registerInputDevice(new SpaceNavigatorDevice);
+#endif
```

Comment:
Same here:
```c++
#if !defined(_USE_3DCONNEXION_SDK)
    ParameterGrp::handle hViewGrp = App::GetApplication().GetParameterGroupByPath(
        "User parameter:BaseApp/Preferences/View");
    if (hViewGrp->GetBool("LegacySpaceMouseDevices", false)) {
        getEventFilter()->registerInputDevice(new SpaceNavigatorDevice);
    }
#else
    getEventFilter()->registerInputDevice(new SpaceNavigatorDevice);
#endif
```

But do we want to register a SpaceNavigatorDevice if we don't have `SPNAV_FOUND`?

Reply:
if we want to restrict this I think it should be `#if defined(_USE_3DCONNEXION_SDK) || defined(SPNAV_FOUND)` like in `GuiApplicationNativeEventAware.cpp`